examples/jsm/utils/BatchedMeshUtils.js

```diff
@@ -0,0 +1,397 @@
+import * as THREE from 'three';
+
+function bufferToHash( buffer ) {
+
+	let hash = 0;
+	if ( buffer.byteLength !== 0 ) {
+
+		let uintArray;
+		if ( buffer.buffer ) {
+
+			uintArray = new Uint8Array(
+				buffer.buffer,
+				buffer.byteOffset,
+				buffer.byteLength
+			);
+
+		} else {
+
+			uintArray = new Uint8Array( buffer );
+
+		}
+
+		for ( let i = 0; i < buffer.byteLength; i ++ ) {
+
+			const byte = uintArray[ i ];
+			hash = ( hash << 5 ) - hash + byte;
+			hash |= 0;
+
+		}
+
+	}
+
+	return hash;
+
+}
+
+/**
+ * Gets material properties hash excluding color (since color can vary within a batch)
+ * @param {THREE.Material} material - The material to hash
+ * @returns {string} Hash of non-color material properties
+ */
+function getMaterialPropertiesHash( material ) {
+
+	const mapProps = [
+		'alphaMap',
+		'aoMap',
+		'bumpMap',
+		'displacementMap',
+		'emissiveMap',
+		'envMap',
+		'lightMap',
+		'metalnessMap',
+		'normalMap',
+		'roughnessMap',
+	];
+
+	// Build map hash
+	const mapHash = mapProps
+		.map( ( prop ) => {
+
+			const map = material[ prop ];
+			if ( ! map ) return 0;
+			return `${map.uuid}_${map.offset.x}_${map.offset.y}_${map.repeat.x}_${map.repeat.y}_${map.rotation}`;
+
+		} )
+		.join( '|' );
+
+	// Build physical properties hash
+	const physicalProps = [
```

Comment:
What if the scene contains instances of `*NodeMaterial` classes (e.g. `MeshStandardNodeMaterial`). 

It looks like properties like `normalNode` or `roughnessNode` are not honored yet.

Reply:
I originally envisioned this utility as a preprocessing step, primarily for 3D assets like GLB files. For MeshStandardNodeMaterial, the material would ideally be applied afterward or the BatchedMesh handled manually, without relying on this utility.

Attempting to integrate the Node system here could lead to maintenance issues and may not be compatible with the WebGLRenderer.